tests/waku_store/test_client.nim

```diff
@@ -0,0 +1,209 @@
+{.used.}
+
+import
+  std/options,
+  testutils/unittests,
+  chronos,
+  chronicles,
+  libp2p/crypto/crypto
+  
+
+import
+  ../../../waku/[
+    node/peer_manager,
+    waku_core,
+    waku_store,
+    waku_store/client,
+  ],
+  ../testlib/[
+    common,
+    wakucore,
+    testasync,
+    futures
+  ],
+  ./store_utils
+
+suite "Store Client":
+  var message1 {.threadvar.}: WakuMessage
+  var message2 {.threadvar.}: WakuMessage
+  var message3 {.threadvar.}: WakuMessage
+  var messageSeq {.threadvar.}: seq[WakuMessage]
+  var handlerFuture {.threadvar.}: Future[HistoryQuery]
+  var handler {.threadvar.}: HistoryQueryHandler
+  var historyQuery {.threadvar.}: HistoryQuery
+
+  var serverSwitch {.threadvar.}: Switch
+  var clientSwitch {.threadvar.}: Switch
+
+  var server {.threadvar.}: WakuStore
+  var client {.threadvar.}: WakuStoreClient
+
+  var serverPeerInfo {.threadvar.}: RemotePeerInfo
+  var clientPeerInfo {.threadvar.}: RemotePeerInfo
+
+  asyncSetup:
+    message1 = fakeWakuMessage(contentTopic=DefaultContentTopic)
+    message2 = fakeWakuMessage(contentTopic=DefaultContentTopic)
+    message3 = fakeWakuMessage(contentTopic=DefaultContentTopic)
+    messageSeq = @[message1, message2, message3]
+    handlerFuture = newHistoryFuture()
+    handler = proc(
+      req: HistoryQuery
+    ): Future[HistoryResult] {.async, gcsafe.} =
+      handlerFuture.complete(req)
+      return ok(HistoryResponse(messages: messageSeq))
+    historyQuery = HistoryQuery(
+      pubsubTopic: some(DefaultPubsubTopic),
+      contentTopics: @[DefaultContentTopic],
+      ascending: true
+    )
+
+    serverSwitch = newTestSwitch()
+    clientSwitch = newTestSwitch()
+
+    server = await newTestWakuStore(serverSwitch, handler=handler)
+    client = newTestWakuStoreClient(clientSwitch)
+
+    await allFutures(serverSwitch.start(), clientSwitch.start())
+
+    serverPeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
+    clientPeerInfo = clientSwitch.peerInfo.toRemotePeerInfo()
+
+  asyncTeardown:
+    await allFutures(serverSwitch.stop(), clientSwitch.stop())
+  
+  suite "HistoryQuery Creation and Execution":
+    asyncTest "Valid Queries":
+      assert false
```

Comment:
What does this assert false do? doesn't it fail the test?

Reply:
That's funny, I can't recall how that ended up there 😂 Might be some piece of code left from testing one bug out. Good catch!